src/MobileRedirectManager.php

```diff
@@ -0,0 +1,305 @@
+	 */
+	public static function is_mobile_request() {
+		/**
+		 * Filters whether the current request is from a mobile device. This is provided as a means to short-circuit
+		 * the normal determination of a mobile request below.
+		 *
+		 * @since 1.6
+		 *
+		 * @param null $is_mobile Whether the current request is from a mobile device.
+		 */
+		$pre_is_mobile = apply_filters( 'amp_pre_is_mobile', null );
+
+		if ( null !== $pre_is_mobile ) {
+			return (bool) $pre_is_mobile;
+		}
+
+		$current_user_agent = wp_unslash( $_SERVER['HTTP_USER_AGENT'] );
+
+		if ( empty( $current_user_agent ) ) {
+			return false;
+		}
+
+		$user_agents_regex = implode(
+			'|',
+			array_map(
+				static function ( $user_agent ) {
+					return preg_quote( $user_agent, '/' );
+				},
+				self::get_user_agents()
+			)
+		);
+
+		if ( preg_match( "/$user_agents_regex/", $current_user_agent ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Determine if mobile redirection should be done via JavaScript.
+	 *
+	 * @return bool True if mobile redirection should be done, false otherwise.
+	 */
+	public static function should_redirect_via_js() {
+		/**
+		 * Filters whether mobile redirection should be done via JavaScript. If false, a server-side solution will be used instead.
+		 *
+		 * @since 1.6
+		 *
+		 * @param bool $should_redirect_via_js Whether JS redirection should be used.
+		 */
+		return (bool) apply_filters( 'amp_redirect_via_js', true );
+	}
+
+	/**
+	 * Get a list of user agents to use for comparison against the user agent from the current request.
+	 *
+	 * @return string[] An array of user agents.
+	 */
+	public static function get_user_agents() {
+		// Default list compiled from the user agents listed in `wp_is_mobile()`.
+		$default_user_agents = [
+			'Mobile',
+			'Android',
+			'Silk/',
+			'Kindle',
+			'BlackBerry',
+			'Opera Mini',
+			'Opera Mobi',
+		];
+
+		/**
+		 * Filters the list of user agents used to determine if the user agent from the current request is a mobile one.
+		 *
+		 * @since 1.6
+		 *
+		 * @param string[] $user_agents List of user agents.
+		 */
+		return apply_filters( 'amp_mobile_user_agents', $default_user_agents );
+	}
+
+	/**
+	 * Determine if mobile redirection is disabled for the current request.
+	 *
+	 * @return bool True if disabled, false otherwise.
+	 */
+	public static function redirection_disabled_for_request() {
+		return ( isset( $_GET[ self::NO_AMP_QUERY_VAR ] ) && '1' === $_GET[ self::NO_AMP_QUERY_VAR ] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	}
+
+	/**
+	 * Determine if mobile redirection is disabled for the browser session.
+	 *
+	 * @return bool True if disabled, false otherwise.
+	 */
+	public static function redirection_disabled_for_session() {
+		return ( isset( $_COOKIE[ self::DISABLED_COOKIE_NAME ] ) && '1' === $_COOKIE[ self::DISABLED_COOKIE_NAME ] ) || self::$disabled_cookie_is_set;
```

Comment:
`FILTER_VALIDATE_BOOLEAN` should be used here, it works with `'1'`, `'true'`, `'on'` and  `'yes'`.

Also, the fastest check should always go first if the order doesn't matter.
```suggestion
		return self::$disabled_cookie_is_set || filter_input( INPUT_GET, self::DISABLED_COOKIE_NAME, FILTER_VALIDATE_BOOLEAN );
```

Reply:
See above.

Also, as opposed to using `filter_input` we could use `rest_sanitize_boolean()`